tests/ha/test_control_pod_restart.py

```diff
@@ -0,0 +1,417 @@
+            LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        if self.restore_node:
+            LOGGER.info("Cleanup: Power on the %s down node.", self.node_name)
+            resp = self.ha_obj.host_power_on(host=self.node_name)
+            assert_utils.assert_true(resp, "Host is not powered on")
+            LOGGER.info("Cleanup: %s is Power on. Sleep for %s sec for pods to join back the"
+                        " node", self.node_name, HA_CFG["common_params"]["pod_joinback_time"])
+            time.sleep(HA_CFG["common_params"]["pod_joinback_time"])
+        # TODO: As cluster restart is not supported until F22A, Need to redeploy cluster after
+        #  every test
+        if self.deploy:
+            LOGGER.info("Cleanup: Destroying the cluster ")
+            resp = self.deploy_lc_obj.destroy_setup(self.node_master_list[0],
+                                                    self.node_worker_list,
+                                                    const.K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Cleanup: Cluster destroyed successfully")
+
+            LOGGER.info("Cleanup: Setting prerequisite")
+            self.deploy_lc_obj.execute_prereq_cortx(self.node_master_list[0],
+                                                    const.K8S_SCRIPTS_PATH,
+                                                    const.K8S_PRE_DISK)
+            for node in self.node_worker_list:
+                self.deploy_lc_obj.execute_prereq_cortx(node, const.K8S_SCRIPTS_PATH,
+                                                        const.K8S_PRE_DISK)
+            LOGGER.info("Cleanup: Prerequisite set successfully")
+
+            LOGGER.info("Cleanup: Deploying the Cluster")
+            resp_cls = self.deploy_lc_obj.deploy_cluster(self.node_master_list[0],
+                                                         const.K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp_cls[0], resp_cls[1])
+            LOGGER.info("Cleanup: Cluster deployment successfully")
+
+        LOGGER.info("Cleanup: Check cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Cleanup: Cluster status checked successfully")
+
+        LOGGER.info("Removing extra files")
+        sysutils.remove_dirs(self.test_dir_path)
+        sysutils.remove_file(self.modified_yaml)
+        self.node_master_list[0].remove_remote_file(self.modified_yaml)
+        self.node_master_list[0].remove_remote_file(self.backup_yaml)
+        LOGGER.info("Done: Teardown completed.")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-40368")
+    @CTFailOn(error_handler)
+    def test_restart_control_node(self):
+        """
+        Verify IOs before and after control pod fails over after restarting node hosting control
```

Comment:
is this test same as that we already developed in pod_failure file?

Reply:
No, added few more checks. Way to failover pod is same as before though.